pennylane/ops/functions/assert_valid.py

```diff
@@ -276,6 +295,42 @@ def _check_bind_new_parameters(op):
         assert qml.math.allclose(d1, d2), failure_comment
 
 
+def _check_differentiation(op):
+    """Checks that the operator can be executed and differentiated correctly."""
+
+    if op.num_params == 0:
+        return
+
+    if isinstance(op, qml.ops.StatePrep):
+        # Fore StatePrep, we do not always expect parameter-shift to agree with backprop
+        # because backprop is unaware of the constraint that the matrix must remain unitary.
+        return
```

Comment:
Do we still need this, or should we just use `skip_differentiation=True`?

Reply:
Let me try removing this

Reply:
I believe there were a number of templates all inheriting from `StatePrep` so I decided to skip this test for all of them all at once here.